src/bp-activity/classes/class-bp-activity-activity.php

```diff
@@ -1573,38 +1610,299 @@ public static function delete_activity_meta_entries( $activity_ids = array() ) {
+	 * @return array The updated activities with nested reactions.
 	 */
-	public static function append_comments( $activities, $spam = 'ham_only' ) {
-		$activity_comments = array();
+	public static function append_reactions( $activities, $spam = 'ham_only' ) {
+		$activity_reactions = array();
 
-		// Now fetch the activity comments and parse them into the correct position in the activities array.
+		// Now fetch the activity reactions and parse them into the correct position in the activities array.
 		foreach ( (array) $activities as $activity ) {
-			$top_level_parent_id = 'activity_comment' == $activity->type ? $activity->item_id : 0;
-			$activity_comments[$activity->id] = BP_Activity_Activity::get_activity_comments( $activity->id, $activity->mptt_left, $activity->mptt_right, $spam, $top_level_parent_id );
+			$top_level_parent_id = 0;
+
+			if ( in_array( $activity->type, bp_get_activity_types_for_role( 'reaction' ), true ) ) {
+				$top_level_parent_id = $activity->item_id;
+			}
+
+			$args = array(
+				'activity_id'         => $activity->id,
+				'mptt_left'           => $activity->mptt_left,
+				'mptt_right'          => $activity->mptt_right,
+				'spam'                => $spam,
+				'top_level_parent_id' => $top_level_parent_id,
+			);
+
+			$function_args = array_values( $args );
+
+			/**
+			 * Filters if BuddyPress should use the legacy activity query.
+			 *
+			 * @since 2.0.0
+			 *
+			 * @param bool                 $value     Whether or not to use the legacy query.
+			 * @param BP_Activity_Activity $value     Magic method referring to currently called method.
+			 * @param array                $func_args Array of the method's argument list.
+			 */
+			if ( 'activity_comment' === $activity->type && apply_filters( 'bp_use_legacy_activity_query', false, 'BP_Activity_Activity::get_activity_comments', $function_args ) ) {
+				list( $activity_id, $left, $right, $spam, $top_level_parent_id ) = $function_args;
+				$activity_reactions[ $activity->id ] = BP_Activity_Activity::get_activity_comments( $activity_id, $left, $right, $spam, $top_level_parent_id );
+
+			} else {
+				$activity_reactions[ $activity->id ] = BP_Activity_Activity::get_activity_reactions( $args );
+			}
 		}
 
-		// Merge the comments with the activity items.
+		// Merge the reactions with the activity items.
 		foreach ( (array) $activities as $key => $activity ) {
-			if ( isset( $activity_comments[$activity->id] ) ) {
-				$activities[$key]->children = $activity_comments[$activity->id];
+			if ( isset( $activity_reactions[$activity->id] ) ) {
+				$activities[ $key ]->children  = $activity_reactions[ $activity->id ]['comments'];
+				$activities[ $key ]->reactions = $activity_reactions[ $activity->id ]['reactions'];
 			}
 		}
 
 		return $activities;
 	}
 
+	/**
+	 * Append activity comments to their associated activity items.
+	 *
+	 * @since 1.2.0
+	 * @deprecated 14.0.0
+	 *
+	 * @global wpdb $wpdb WordPress database object.
```

Comment:
This can be removed.

Reply:
Ok, good to know. I'll leave it, just like WordPress does.